kclvm/parser/src/entry.rs

```diff
@@ -0,0 +1,413 @@
+///         .unwrap()
+///         .display()
+///         .to_string(),
+///     testpath
+///         .join("kcl2")
+///         .canonicalize()
+///         .unwrap()
+///         .to_str()
+///         .unwrap()
+/// );
+///
+/// assert_eq!(entries.get_nth_entry(2).unwrap().name(), "__main__");
+/// assert_eq!(
+///     PathBuf::from(entries.get_nth_entry(2).unwrap().path())
+///         .display()
+///         .to_string(),
+///     kcl1_path.join("main.k").canonicalize().unwrap().to_str().unwrap()
+/// );
+///
+/// ```
+pub fn get_compile_entries_from_paths(
+    file_paths: &[String],
+    opts: &LoadProgramOptions,
+) -> Result<Entries, String> {
+    if file_paths.is_empty() {
+        return Err("No input KCL files or paths".to_string());
+    }
+    let mut result = Entries::default();
+    let mut m = std::collections::HashMap::<String, String>::new();
+    for s in file_paths {
+        let path = ModRelativePath::from(s.to_string());
+        // If the path is a [`ModRelativePath`] with preffix '${<package_name>:KCL_MOD}',
+        // calculate the real path and the package name.
+        if let Some((pkg_name, pkg_path)) = path
+            .get_root_pkg_name()
+            .map_err(|err| err.to_string())?
+            .and_then(|name| {
+                opts.package_maps
+                    .get(&name)
+                    .map(|pkg_path: &String| (name, pkg_path))
+            })
+        {
+            // Replace the mod relative path preffix '${<pkg_name>:KCL_MOD}' with the real path.
+            let s = path
+                .canonicalize_by_root_path(pkg_path)
+                .map_err(|err| err.to_string())?;
+            if let Some(root) = get_pkg_root(&s) {
+                let mut entry = Entry::new(pkg_name.clone(), root.clone());
+                entry.extend_k_files(get_main_files_from_pkg_path(&s, &root, &pkg_name, opts)?);
+                result.push_entry(entry);
+                continue;
+            }
+            // If the [`ModRelativePath`] with preffix '${KCL_MOD}'
+        } else if path.is_relative_path().map_err(|err| err.to_string())?
+            && path
+                .get_root_pkg_name()
+                .map_err(|err| err.to_string())?
+                .is_none()
+        {
+            // Push it into `result`, and deal it later.
+            result.push(kclvm_ast::MAIN_PKG.to_string(), path.get_path());
+            continue;
+        } else if let Some(root) = get_pkg_root(s) {
+            // If the path is a normal path.
+            m.insert(root.clone(), root.clone());
+            let mut entry = Entry::new(kclvm_ast::MAIN_PKG.to_string(), root.clone());
+            entry.extend_k_files(get_main_files_from_pkg_path(
+                &s,
+                &root,
+                &kclvm_ast::MAIN_PKG.to_string(),
+                opts,
+            )?);
+            result.push_entry(entry);
+        }
+    }
+
+    if m.is_empty() {
+        m.insert("".to_string(), "".to_string());
```

Comment:
What does the statement mean?

Reply:
This is the code debt before, I copied. I removed it.